examples/backpack-demo/index.html

```diff
@@ -0,0 +1,156 @@
+      <value name="TIMES">
+        <shadow type="math_number">
+          <field name="NUM">10</field>
+        </shadow>
+      </value>
+    </block>
+    <block type="controls_flow_statements"></block>
+  </category>
+  <category name="Math" categorystyle="math_category">
+    <block type="math_number" gap="32">
+      <field name="NUM">123</field>
+    </block>
+    <block type="math_arithmetic">
+      <value name="A">
+        <shadow type="math_number">
+          <field name="NUM">1</field>
+        </shadow>
+      </value>
+      <value name="B">
+        <shadow type="math_number">
+          <field name="NUM">1</field>
+        </shadow>
+      </value>
+    </block>
+    <block type="math_single">
+      <value name="NUM">
+        <shadow type="math_number">
+          <field name="NUM">9</field>
+        </shadow>
+      </value>
+    </block>
+    <block type="math_number_property">
+      <value name="NUMBER_TO_CHECK">
+        <shadow type="math_number">
+          <field name="NUM">0</field>
+        </shadow>
+      </value>
+    </block>
+  </category>
+  <category name="Text" categorystyle="text_category">
+    <block type="text"></block>
+    <block type="text_multiline"></block>
+    <label text="Input/Output:" web-class="ioLabel"></label>
+    <block type="text_print">
+      <value name="TEXT">
+        <shadow type="text">
+          <field name="TEXT">abc</field>
+        </shadow>
+      </value>
+    </block>
+    <block type="text_prompt_ext">
+      <value name="TEXT">
+        <shadow type="text">
+          <field name="TEXT">abc</field>
+        </shadow>
+      </value>
+    </block>
+  </category>
+  <sep></sep>
+  <category name="Variables" categorystyle="variable_category" custom="VARIABLE"></category>
+  <category name="Functions" categorystyle="procedure_category" custom="PROCEDURE"></category>
+</xml>
+
+<script>
+  // Inject primary workspace.
+  var primaryWorkspace = Blockly.inject('primaryDiv',
+      {
+        media: 'https://unpkg.com/blockly/media/',
+        toolbox: document.getElementById('toolbox'),
+        trashcan: true,
+      });
+  // Inject secondary workspace.
+  var secondaryWorkspace = Blockly.inject('secondaryDiv',
+      {
+        media: 'https://unpkg.com/blockly/media/',
+        toolbox: document.getElementById('toolbox'),
+        trashcan: true
+      });
+
+  // Add backpacks
+  const backpack1 = new Backpack(primaryWorkspace);
+  backpack1.init();
+  const backpack2 = new Backpack(secondaryWorkspace);
+  backpack2.init();
+
+  // Listen to events on both workspace.
+  primaryWorkspace.addChangeListener(updateBackpack);
+  secondaryWorkspace.addChangeListener(updateBackpack);
+
+  function updateBackpack(event) {
+    if (event.type !== 'backpack_change') {
+      return;
+    }
+    Blockly.Events.disable();
+    let contents;
+    let targetBackpack;
+    if (primaryWorkspace.id === event.workspaceId) {
+      targetBackpack = secondaryWorkspace.backpack;
```

Comment:
If this is a monkey patch that we expect to remove at some point, could we instead use `backpack1` and `backpack2` here instead of getting it from the workspace?